src/dotnet/commands/dotnet-build/Program.cs

```diff
@@ -93,10 +105,55 @@ public static int Run(string[] args)
 
                 msbuildArgs.AddRange(app.RemainingArguments);
 
-                return new MSBuildForwardingApp(msbuildArgs).Execute();
+                ret._forwardingApp = new MSBuildForwardingApp(msbuildArgs, msbuildPath);
+
+                return 0;
             });
 
-            return app.Execute(args);
+            int exitCode = app.Execute(args);
+            if (ret._forwardingApp == null)
+            {
+                throw new CodeNotExecutedException(exitCode);
+            }
+
+            return ret;
+        }
+
+        public static int Run(string[] args)
+        {
+            DebugHelper.HandleDebugSwitch(ref args);
+
+            BuildCommand cmd;
+            try
+            {
+                cmd = FromArgs(args);
+            }
+            catch (CodeNotExecutedException e)
```

Comment:
I think the confusion is because cmd line parser is running some code as part of parsing. This is likely misnamed exception but I got no better idea for the name - any ideas?